### PR TITLE
raise custom exception and catch it if catalog too big to page

### LINKED ISSUE #51: BUG: Could not page to end of big catalog
Loaded a catalog with 88k runs.  The first 20 runs displayed in the results table.  After pressing the `last` pager button, the app crashed with this exception trace:

```py
(bluesky_2023_2) jemian@otz ~/.../gemviz23/demo $ catalogSelected: args = ('developer',)  kwargs = {}
Displaying catalog: developer
catalogSelected: args = ('20idb_usaxs',)  kwargs = {}
Displaying catalog: 20idb_usaxs
action='next' kwargs={}
doPager action ='next', value =None
self.pageOffset() =20 self.pageSize() =20
action='last' kwargs={}
doPager action ='last', value =None
Traceback (most recent call last):
  File "/home/beams1/JEMIAN/Documents/projects/BCDA-APS/gemviz23/gemviz23/demo/resultwindow.py", line 186, in doPagerButtons
    model.doPager(action)
  File "/home/beams1/JEMIAN/Documents/projects/BCDA-APS/gemviz23/gemviz23/demo/resultwindow.py", line 83, in doPager
    self.setUidList(self._get_uidList())
  File "/home/beams1/JEMIAN/Documents/projects/BCDA-APS/gemviz23/gemviz23/demo/resultwindow.py", line 97, in _get_uidList
    return list(gen)
  File "/home/beams/JEMIAN/.conda/envs/bluesky_2023_2/lib/python3.10/site-packages/tiled/client/node.py", line 352, in _keys_slice
    content = self.context.get_json(
  File "/home/beams/JEMIAN/.conda/envs/bluesky_2023_2/lib/python3.10/site-packages/tiled/client/context.py", line 595, in get_json
    self.get_content(
  File "/home/beams/JEMIAN/.conda/envs/bluesky_2023_2/lib/python3.10/site-packages/tiled/client/context.py", line 557, in get_content
    handle_error(response)
  File "/home/beams/JEMIAN/.conda/envs/bluesky_2023_2/lib/python3.10/site-packages/tiled/client/utils.py", line 29, in handle_error
    response.raise_for_status()
  File "/home/beams/JEMIAN/.conda/envs/bluesky_2023_2/lib/python3.10/site-packages/httpx/_models.py", line 749, in raise_for_status
    raise HTTPStatusError(message, request=request, response=self)
httpx.HTTPStatusError: Server error '500 Internal Server Error' for url 'http://otz.xray.aps.anl.gov:8000/api/v1/node/search/20idb_usaxs?page%5Boffset%5D=88196&fields=&sort=time'
For more information check: https://httpstatuses.com/500
```

### PR DESCRIPTION
**Pull request by prjemian**
- close #51

### CODE REVIEW

File: gemviz/bluesky_runs_catalog_table_view.py [hunk 1]
```diff
@@ -55,10 +55,13 @@ def doPagerButtons(self, action, **kwargs):
...
-            self.setStatus(f"{model.pageOffset()=}")
-        self.doButtonPermissions()
-        self.setPagerStatus()
+            try:
+                model.doPager(action)
+                self.setStatus(f"{model.pageOffset()=}")
+                self.doButtonPermissions()
+                self.setPagerStatus()
+            except utils.TiledServerError as exc:
+                self.setStatus(str(exc))
```
Comment (prjemian) [inline gemviz/bluesky_runs_catalog_table_view.py:64]: Could raise a modal dialog about this.

**Comment by prjemian** [comment]
Confirmed with @danielballan that paging larger catalogs is a limitation of an early decision in databroker use of MongoDM.  There is a [fix possible](https://github.com/bluesky/suitcase-mongo/pull/49/files) for any catalog that presents this problem.  The fix, which relies on use of the [`mongosh`](https://www.mongodb.com/docs/mongodb-shell/) program, adds additional indices that optimize paging through larger catalogs.

```bash
$ mongosh "${CATALOG_URI}"
db.run_start.createIndex({"scan_id": -1, "_id": -1}, {unique: true})
db.run_start.createIndex({"time": -1, "_id": -1}, {unique: true})
db.run_start.createIndex({"scan_id": 1, "_id": -1}, {unique: true})
db.run_start.createIndex({"scan_id": 1}, {unique: false})
```

Eventually, this fix will be pushed to a release so that any (and all) new catalogs will be created with these indices.

**Comment by prjemian** [comment]
Obviously, that change to a MongoDB catalog should be done only by someone who has the skills to undo it if a problem occurs.

**Comment by prjemian** [comment]
Adding a modal dialog to show the message:

image

**Comment by prjemian** [comment]
@rodolakis Ready for review

**Review by rodolakis — APPROVED** [review]
LGTM

**Pull request merged**